File: java/src/main/java/de/sven_torben/serialization_benchmark/testdata/java/Catalog.java
package de.sven_torben.serialization_benchmark.testdata.java;

import java.io.Serializable;


public final class Catalog implements Serializable {

	private static final long serialVersionUID = 1L;
	
	private CatalogItem[] items;
	
	public Catalog(){
		this(new CatalogItem[0]);
	}
	
	public Catalog(final CatalogItem[] items) {
		this.items = items;
	}
	
	public void setItems(final CatalogItem[] items)
	{
		this.items = items;
	}
	
	public CatalogItem[] getItems()
	{
		return items;
	}

}

File: net/Testdata/Catalog.cs
//using Ch.Elca.Iiop.Idl;
using ProtoBuf;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Runtime.Serialization;
using System.Text;
using System.Threading.Tasks;
using System.Xml.Serialization;

namespace De.Sven_Torben.Serialization_Benchmark.Testdata
{
    [Serializable]
    [ProtoContract]
    [DataContract]
    [XmlRoot]
    [XmlType]
    //[ImplClass("MessageTypeComparison.CatalogImpl"), RepositoryID("IDL:MessageTypeComparison/Catalog:1.0")]
    public class Catalog
    {

        [ProtoMember(1, DataFormat = DataFormat.Group)]
        [DataMember]
        [XmlArray]
        public CatalogItem[] Items { get; set; }
    }
}
